File: aspnet-in-practice/15/VB/VirtualPathProvider/App_Code/DatabaseDirectory.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Hosting
Imports System.Collections

Namespace ASPNET4InPractice
	Public Class DatabaseDirectory
		Inherits VirtualDirectory
		Private _directories As New List(Of String)()
		Private _files As New List(Of String)()
		Private _children As New List(Of String)()

		Public Sub New(ByVal virtualPath As String)
			MyBase.New(virtualPath)
		End Sub

		Public Overloads Overrides ReadOnly Property Children() As IEnumerable
			Get
				Return _children
			End Get
		End Property

		Public Overloads Overrides ReadOnly Property Directories() As IEnumerable
			Get
				Return _directories
			End Get
		End Property

		Public Overloads Overrides ReadOnly Property Files() As IEnumerable
			Get
				Return _files
			End Get
		End Property
	End Class
End Namespace

File: aspnet-in-practice/15/CS/VirtualPathProvider/App_Code/DatabaseDirectory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Hosting;
using System.Collections;

namespace ASPNET4InPractice
{
	public class DatabaseDirectory : VirtualDirectory
	{
		private List<string> _directories = new List<string>();
		private List<string> _files = new List<string>();
		private List<string> _children = new List<string>();

		public DatabaseDirectory(string virtualPath): base(virtualPath)	{}

		public override IEnumerable Children
		{
			get
			{
				return _children;
			}
		}

		public override IEnumerable Directories
		{
			get
			{
				return _directories;
			}
		}

		public override IEnumerable Files
		{
			get
			{
				return _files;
			}
		}
	}
}